specs/WorkerPostMessage.md

```diff
@@ -0,0 +1,682 @@
+    std::wstring url, wil::com_ptr<ICoreWebView2ServiceWorker> serviceWorker)
+{
+    std::wstring msg = L"{\"command\":\"ADD_TO_CACHE\",\"url\":\"" + url + L"\"}";
+    //! [PostWebMessageAsJson]
+    serviceWorker->PostWebMessageAsJson(msg.c_str());
+    //! [PostWebMessageAsJson]
+}
+```
+
+### .NET/WinRT
+```c#
+CoreWebView2ServiceWorkerManager PostMessage_ServiceWorkerManager_;
+
+void ServiceWorkerPostMessageExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    PostMessage_ServiceWorkerManager_ = WebViewProfile.ServiceWorkerManager;
+
+    PostMessage_ServiceWorkerManager_.ServiceWorkerRegistered +=
+        ServiceWorker_PostMessage_ServiceWorkerRegistered;
+}
+
+void ServiceWorker_PostMessage_ServiceWorkerRegistered(object sender,
+        CoreWebView2ServiceWorkerRegisteredEventArgs args)
+{
+    CoreWebView2ServiceWorkerRegistration serviceWorkerRegistration =
+        args.ServiceWorkerRegistration;
+    MessageBox.Show("Service worker is registered for " + serviceWorkerRegistration.ScopeUri,
+                    "Service Worker Registration Message");
+
+    CoreWebView2ServiceWorker serviceWorker = serviceWorkerRegistration.ActiveServiceWorker;
+    if (serviceWorker != null)
+    {
+        ServiceWorker_PostMessage_SetupEventsOnServiceWorker(serviceWorker);
+        ServiceWorker_PostMessage_AddToCache(serviceWorker, "img.jpg");
+    }
+    else
+    {
+        serviceWorkerRegistration.ServiceWorkerActivated += (sender1, args1) =>
+        {
+            ServiceWorker_PostMessage_SetupEventsOnServiceWorker(serviceWorker);
+            ServiceWorker_PostMessage_AddToCache(serviceWorker, "img.jpg");
+        };
+    }
+}
+
+void ServiceWorker_PostMessage_SetupEventsOnServiceWorker(CoreWebView2ServiceWorker serviceWorker)
+{
+    serviceWorker.WebMessageReceived += (sender, args) =>
+    {
+        StringBuilder messageBuilder = new StringBuilder();
+        messageBuilder.AppendLine($"Service Worker: \n{args.Source} ");
+        messageBuilder.AppendLine($"\nMessage: \n{args.TryGetWebMessageAsString()} ");
+        MessageBox.Show(messageBuilder.ToString(), "Message from Service Worker",
+                        MessageBoxButton.OK);
+    };
+}
+
+void ServiceWorker_PostMessage_AddToCache(CoreWebView2ServiceWorker serviceWorker, string url)
+{
+    string msg = "{\"command\":\"ADD_TO_CACHE\",\"url\":\"" + url + "\"}";
+    serviceWorker.PostWebMessageAsJson(msg);
+}
+```
+
+# API Details
+## C++
+```
+/// Receives `WebMessageReceived` events.
+[uuid(b366218b-0bb8-58a3-ac33-f40a2235366e), object, pointer_default(unique)]
+interface ICoreWebView2DedicatedWorkerWebMessageReceivedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2DedicatedWorker* sender,
+      [in] ICoreWebView2WebMessageReceivedEventArgs* args);
+}
+
+[uuid(66833876-edba-5a60-8508-7da64504a9d2), object, pointer_default(unique)]
+interface ICoreWebView2DedicatedWorker : IUnknown {

```

Comment:
I see there is a CoreWebView2DedicatedWorker in .net prerelease.  This interface doesn't need a "2", just checking is that because the class hasn't gone to stable release yet?